### PR TITLE
Add GCP oslogin authentication

### LINKED ISSUE #970: ray up fails when `enable-oslogin` set in GCP
When the `enable-oslogin` is set to True in the metadata of GCP project, `ray up` will fail due to the process fail to connect to the instance with `ssh` and the key added to the `metadata/ssh_keys`, as stated in the [GCP document](https://cloud.google.com/compute/docs/instances/ssh#os_login-managed_ssh_connections): 
> When OS Login is enabled, Compute Engine refuses connections from SSH keys that are stored in metadata.
image

This seems a ray problem with the latest GCP permission system. We may either:
1. add instructions for the users to remove the option from their project in our document; or
2. modify the gcp node_provider ourselves.

### PR DESCRIPTION
**Pull request by Michaelvll**
Closes #970.

This PR enables running sky in the GCP project with oslogin enabled.

Tested:
- [x] `sky launch -c test-gcp --cloud gcp ''`
- [x] `sky launch -c test-custom-gcp launch.yaml`
```
# launch.yaml
resources:
  instance_type: n2d-highmem-8
  region: us-west1
  image_id: projects...
run: |
  echo hi
```
- [x] `sky down -a`

### CODE REVIEW

File: sky/skylet/ray_patches/__init__.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def _run_patch(target_file, patch_file):
     # .orig is the original file that is not patched.
     orig_file = os.path.abspath(target_file + '.orig')
     script = f"""\
+    which patch >/dev/null 2>&1 || sudo yum install -y patch
```
Comment (concretevitamin) [inline sky/skylet/ray_patches/__init__.py:31]: `yum` may not be available on all images. Should we catch this gracefully? E.g., `|| true` at end of L31, and on a new command check `which patch` is available and if not exit early?

**Review by concretevitamin — APPROVED** [review]
Very nice @Michaelvll. Should we run smoke tests too before merging?

File: sky/skylet/ray_patches/__init__.py (hunk 1 above)
Comment (Michaelvll) [reply sky/skylet/ray_patches/__init__.py]: Ahh, good catch! Just added. Thanks!

**Comment by Michaelvll** [comment]
Tested:
- [ ] `tests/run_smoke_tests.sh`

**Pull request merged**